qiskit_dynamics/models/rotating_frame.py

```diff
@@ -337,11 +336,8 @@ def _conjugate_and_add(
             if op_to_add_in_fb is None:
                 return operator
             else:
-                if op_to_add_in_fb is not None:
-                    if issparse(operator):
-                        op_to_add_in_fb = csr_matrix(op_to_add_in_fb)
-                    elif type(operator).__name__ == "BCOO":
-                        op_to_add_in_fb = to_BCOO(op_to_add_in_fb)
+                if issparse(operator) or type(operator).__name__ == "BCOO":
+                    op_to_add_in_fb = DYNAMICS_NUMPY_ALIAS(like=operator).asarray(op_to_add_in_fb)
```

Comment:
Do we need the if statement checking sparse typee here or can we just do:

```suggestion
                op_to_add_in_fb = DYNAMICS_NUMPY_ALIAS(like=operator).asarray(op_to_add_in_fb)
```